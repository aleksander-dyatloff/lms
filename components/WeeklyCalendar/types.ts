import ChangeHandler from '@interfaces/ChangeHandler'
import Timestamp from '@interfaces/Timestamp'

namespace WeeklyCalendarComponent {
  export interface WrapperProps {
  }

  export interface Props {
    value: Timestamp
    onChange: ChangeHandler
  }
}

export default WeeklyCalendarComponent
