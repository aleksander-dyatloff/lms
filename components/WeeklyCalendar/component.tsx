import Divider from '@components/Divider'
import Text from '@components/Text'
import { FC } from 'react'

import Wrapper from './styles'
import WeeklyCalendarComponent from './types'

const WeeklyCalendar: FC<WeeklyCalendarComponent.Props> = ({
  onChange,
  value,
  ...restProps
}) => (
  <Wrapper
    {...restProps}
  >
    <div className='head'>
      <div className='headItem'>
        <Text as='sub'>
          Mon
        </Text>
        <Text
          className='dayNumber active'
          variant='h6'
        >
          7
        </Text>
      </div>
      <div className='headItem'>
        <Text as='sub'>
          Tue
        </Text>
        <Text
          className='dayNumber'
          variant='h6'
        >
          8
        </Text>
      </div>
      <div className='headItem'>
        <Text as='sub'>
          Wed
        </Text>
        <Text
          className='dayNumber'
          variant='h6'
        >
          9
        </Text>
      </div>
      <div className='headItem'>
        <Text as='sub'>
          Thu
        </Text>
        <Text
          className='dayNumber'
          variant='h6'
        >
          10
        </Text>
      </div>
      <div className='headItem'>
        <Text as='sub'>
          Fri
        </Text>
        <Text
          className='dayNumber'
          variant='h6'
        >
          11
        </Text>
      </div>
      <div className='headItem'>
        <Text as='sub'>
          Sat
        </Text>
        <Text
          className='dayNumber'
          variant='h6'
        >
          12
        </Text>
      </div>
      <div className='headItem'>
        <Text as='sub'>
          Sun
        </Text>
        <Text
          className='dayNumber'
          variant='h6'
        >
          13
        </Text>
      </div>
    </div>
    <Divider />
    <div className='body'>
      <div className='bodyItem'>
        <div className='panel'>
          <div className='panelHeader'>
            <Text>English</Text>
          </div>
          <Text variant='sub'>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Corrupti, optio?
          </Text>
        </div>
      </div>
      <div className='bodyItem' />
      <div className='bodyItem' />
      <div className='bodyItem' />
      <div className='bodyItem' />
      <div className='bodyItem' />
      <div className='bodyItem' />
    </div>
  </Wrapper>
)

export default WeeklyCalendar
