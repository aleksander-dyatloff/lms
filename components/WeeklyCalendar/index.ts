import WeeklyCalendar from './component'

export default WeeklyCalendar
