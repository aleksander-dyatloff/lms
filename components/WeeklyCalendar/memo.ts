import { memo } from 'react'

import WeeklyCalendar from './component'

export default memo(WeeklyCalendar)
