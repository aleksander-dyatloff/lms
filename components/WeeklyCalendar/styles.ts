import styled from 'styled-components'

import WeeklyCalendarComponent from './types'

const Wrapper = styled.div<WeeklyCalendarComponent.WrapperProps>`
  width: 100%;
  height: 100%;
  background-color: #f3f3f3;
  border-radius: ${({ theme }) => theme.borderRadius * 4}px;

  .head {
    display: flex;
    align-items: center;
    padding: ${({ theme }) => theme.spacing * 4}px;
  }

  .headItem, .bodyItem {
    flex: 1 0;
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
  }

  .dayNumber {
    padding: ${({ theme }) => theme.spacing}px;
    min-width: 40px;
    min-height: 40px;
    width: auto;
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 50%;

    &.active {
      background-color: ${({ theme }) => theme.palette.primary.main};
      color: ${({ theme }) => theme.palette.primary.contrast};
    }
  }

  .body {
    display: flex;
    padding: ${({ theme }) => theme.spacing * 4}px;
  }

  .panel {
    background-color: ${({ theme }) => theme.palette.primary.contrast};
    border-radius: ${({ theme }) => theme.borderRadius * 2}px;
    overflow: hidden;

    & > * {
      padding: ${({ theme }) => theme.spacing * 1}px;
    }
  }

  .panelHeader {
    background-color: #64a3f5;
    color: white;
  }
`
export default Wrapper
