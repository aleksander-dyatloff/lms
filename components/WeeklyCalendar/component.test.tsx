import 'jest-styled-components'

import { render } from '@testing-library/react'
import { Profiler } from 'react'

import WeeklyCalendar from './component'

describe('WeeklyCalendar component', () => {
  test('should match to snapshot', () => {
    const { container } = render(<WeeklyCalendar />)

    expect(container).toMatchSnapshot()
  })

  test('first render duration should match to specification', () => {
    render(
      <Profiler
        id='WeeklyCalendar'
        onRender={(_, __, renderDuration) => {
          expect(renderDuration).toBeLessThanOrEqual(1.5)
        }}
      >
        <WeeklyCalendar />
      </Profiler>,
    )
  })
})
