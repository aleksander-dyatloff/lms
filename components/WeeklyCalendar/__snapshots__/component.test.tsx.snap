// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeeklyCalendar component should match to snapshot 1`] = `
<div>
  <div
    class=""
  />
</div>
`;
